hyperapp.d.ts

```diff
@@ -0,0 +1,45 @@
+export as namespace hyperapp
+
+// VNode part
+
+export interface VNode<Data> {
+  tag: string
+  data?: Data
+  children: VElement<{}>[]
+}
+
+export type VElement<Data> = VNode<Data> | string
+
+export interface Component<Data> {
+  (data?: Data, ...children: VElement<{}>[]): VNode<Data>
+}
+
+export function h<Data>(
+  tag: Component<Data> | string,
+  data?: Data,
+  children?: VElement<{}>[]
+): VNode<Data>
+export function h<Data>(
+  tag: Component<Data> | string,
+  data?: Data,
+  ...children: VElement<{}>[]
+): VNode<Data>
+
```

Comment:
Only in `h()` or we can tell a `VirtualElement` could be either a `VirtualNode`, a `string`, or a `number`.

The last one will affect : 
* `h()` children parameter
* `Component` children
* `VirtualNode` children